.github/workflows/build-and-push-Docker-image.yml

```diff
@@ -10,33 +12,42 @@ on:
     outputs:
       full-image-name:
         description: "Full pushed image name including host/registry, name, and tag"
-        value: ${{ jobs.docker-build.outputs.full-image-name }}
+        value: ${{ jobs.docker.outputs.full-image-name }}
+
 jobs:
-  docker-build:
+  docker:
     runs-on: ubuntu-latest
     permissions:
       packages: write
       contents: read
     timeout-minutes: 25
     outputs:
-      full-image-name: ${{ steps.push-image.outputs.full-image-name }}
+      full-image-name: ${{ steps.image-name.outputs.full-image-name }}
 
     steps:
-      - name: Checkout
-        # https://github.com/actions/checkout
-        uses: actions/checkout@v3
-
-      - name: Build Docker image
-        run: docker build . --file Dockerfile --tag ${{ inputs.image-name }}
-
-      - name: Log in to registry
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
 
-      - name: Push image
-        id: push-image
+      - name: Define image name
+        id: image-name
         run: |
           full_image_name="ghcr.io/${{ github.repository_owner }}/${{ inputs.image-name }}:${{ inputs.image-tag }}"
           full_image_name=$(echo $full_image_name | tr '[A-Z]' '[a-z]')
-          docker tag ${{ inputs.image-name }} $full_image_name
-          docker push $full_image_name
-          echo "full-image-name=$full_image_name" >> $GITHUB_OUTPUT
+          echo "full-image-name=$full_image_name" >> "$GITHUB_OUTPUT"
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
```

Comment:
I went down this rabbit hole 6 months ago but abandoned it... if you can get this working that's great! It would be awesome if both an amd64 and an arm64 image get pushed so that when you pull it locally you automatically get the most appropriate version for the platform you're running on.

Reply:
That's the general plan. First step is to get buildx working, and then worry about that. #20